ext/re2/extconf.rb

```diff
@@ -28,6 +29,28 @@
   "/usr/lib"
 ]
 
+def libflag_to_filename(ldflag)
+  case ldflag
+  when /\A-l(.+)/
+    "lib#{Regexp.last_match(1)}.#{$LIBEXT}"
+  end
+end
+
+def resolve_static_library(arg)
+  libs_path = pkg_config('re2', 'libs-only-L')
```

Comment:
On Linux, this appears to return an empty list.

Reply:
It looks like the static library isn't shipped in the `libre2-5` package.

Reply:
However, the static library is shipped in `libre2-dev` under `/usr/lib/x86_64-linux-gnu/libre2.a`.

Reply:
Hmm, still getting:

```
g++ -shared -o re2.so re2.o -L. -L/home/stanhu/.rbenv/versions/2.7.5/lib -Wl,-rpath,/home/stanhu/.rbenv/versions/2.7.5/lib -L/usr/local/lib -Wl,-rpath,/usr/local/lib -L/opt/homebrew/lib -Wl,-rpath,/opt/homebrew/lib -L/usr/lib -Wl,-rpath,/usr/lib -L/usr/lib/x86_64-linux-gnu -Wl,-rpath,/usr/lib/x86_64-linux-gnu -L. -L/home/stanhu/.rbenv/versions/2.7.5/lib  -fstack-protector-strong -rdynamic -Wl,-export-dynamic -L/home/stanhu/.rbenv/versions/2.7.5/lib  -Wl,--compress-debug-sections=zlib    -Wl,-rpath,/home/stanhu/.rbenv/versions/2.7.5/lib -L/home/stanhu/.rbenv/versions/2.7.5/lib -lruby /usr/lib/x86_64-linux-gnu/libre2.a -lstdc++ -lm   -lc
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libre2.a(re2.o): relocation R_X86_64_PC32 against symbol `_ZTVSt9basic_iosIcSt11char_traitsIcEE@@GLIBCXX_3.4' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
```

Reply:
Could be related to Ubuntu's compilation: https://github.com/google/re2/issues/178

Reply:
Perhaps we're better off doing https://github.com/mudge/re2/issues/61.

Reply:
Confirmed that the installed Ubuntu shared library appears to be missing `-fPIC` in some objects. Recompiling with `CPPFLAGS=-fPIC make` seems to work.